php-checks/src/main/java/org/sonar/php/checks/ClassNameCheck.java

```diff
@@ -16,23 +16,29 @@
  */
 package org.sonar.php.checks;
 
-import java.util.Collections;
-import java.util.List;
+import java.util.Map;
 import java.util.regex.Pattern;
 import org.sonar.check.Rule;
 import org.sonar.check.RuleProperty;
+import org.sonar.plugins.php.api.symbols.SymbolTable;
+import org.sonar.plugins.php.api.tree.ScriptTree;
 import org.sonar.plugins.php.api.tree.Tree;
 import org.sonar.plugins.php.api.tree.declaration.ClassDeclarationTree;
 import org.sonar.plugins.php.api.tree.expression.NameIdentifierTree;
-import org.sonar.plugins.php.api.visitors.PHPSubscriptionCheck;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
 
-@Rule(key = ClassNameCheck.KEY)
-public class ClassNameCheck extends PHPSubscriptionCheck {
+@Rule(key = "S101")
+public class ClassNameCheck extends PHPVisitorCheck {
 
-  public static final String KEY = "S101";
-  private static final String MESSAGE = "Rename class \"%s\" to match the regular expression %s.";
+  private static final String MESSAGE = "Rename %s \"%s\" to match the regular expression %s.";
+  private static final Map<Tree.Kind, String> DECLARATION_TYPE_LABELS = Map.of(
```

Comment:
Was it discussed that we want this rule to also raise on interfaces and traits? 
Personally, I would say it makes sense. However it feels strange having a ticket called `S101 should not raise for classes generated by Yii` which does this change on the side. I would have expected this PR not to add new findings. 

I suggest either doing the change of additionally raising on interfaces and traits in a subsequent PR/Ticket or adapting SONARPHP-1674 to make clear that it is also about this.

Reply:
@ZiadMarey can you update the ticket description?